### PR TITLE
Higher order (second) Sobol indices and Morris update 

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**

### CODE REVIEW

File: src/morris_sensitivity.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@
 end
 
 struct MatSpread
-    mat::Vector{Vector{Float64}}
+    mat::AbstractMatrix
```
Comment (ChrisRackauckas) [inline src/morris_sensitivity.jl]: parameterize

File: test/morris_method.jl [hunk 2]
```diff
@@ -4,12 +4,22 @@ A = reshape([1,0,2,3],2,2)
...
-
-m = gsa(f_morris,Morris(p_steps=[10,10],relative_scale=true,len_trajectory=2000,total_num_trajectory=1000,num_trajectory=150),[[1,5],[1,5]])
-@test m.means[1][2] ≈ 0 atol=1e-12
-@test m.variances[1][2] ≈ 0 atol=1e-12
-@test m.means[2][1] < m.means[2][2]
+
+function linear_batch(X)
+    A= 7
+    B= 0.1
+    @. A*X[1,:]+B*X[2,:]
```
Comment (ChrisRackauckas) [inline test/morris_method.jl]: it also needs to handle multi-output?

Comment (Vaibhavdixit02) [reply test/morris_method.jl]: It does?

File: test/morris_method.jl [hunk 3]
```diff
@@ -4,12 +4,31 @@ A = reshape([1,0,2,3],2,2)
...
+end
+
+function ishi_linear(X)
+    A= 7
+    B= 0.1
+    [sin(X[1]) + A*sin(X[2])^2+ B*X[3]^4 *sin(X[1]),A*X[1]+B*X[2]]
+end
```
Comment (ChrisRackauckas) [inline test/morris_method.jl]: Missing the ishi_linear_batch test

File: test/morris_method.jl [hunk 4]
```diff
@@ -4,12 +4,45 @@ A = reshape([1,0,2,3],2,2)
...
+@test m.means[:,2] ≈ A[:,2] atol=1e-12
+@test m.variances ≈ reshape([0,0,0,0],2,2) atol=1e-12
+
+m = gsa(f_morris,Morris(p_steps=[10,10],relative_scale=true,total_num_trajectory=1000,num_trajectory=150),[[1,5],[1,5]])
+@test m.means[2,1] ≈ 0 atol=1e-12
+@test m.variances[2,1] ≈ 0 atol=1e-12
+@test m.means[1,2] < m.means[2,2]
+
+m = gsa(linear_batch,Morris(p_steps=[10,10],relative_scale=true,num_trajectory=10000),[[1,5],[1,5]],batch=true)
+@test m.means ≈ [0.439769  0.00579919] atol = 1e-2
```
Comment (ChrisRackauckas) [inline test/morris_method.jl]: linear and linear batch should give the same result.

File: test/morris_method.jl [hunk 5]
```diff
@@ -4,12 +4,45 @@ A = reshape([1,0,2,3],2,2)
...
+
+m = gsa(ishi_linear,Morris(p_steps=[10,10],relative_scale=false,num_trajectory=100),[[lb[i],ub[i]] for i in 1:4])
```
Comment (ChrisRackauckas) [inline test/morris_method.jl]: no tests on ishi?

File: test/morris_method.jl (hunk 4 above)
Comment (Vaibhavdixit02) [reply test/morris_method.jl]: That was because of `relative_scale` (on in one off in other). Fixed.

File: test/morris_method.jl (hunk 5 above)
Comment (Vaibhavdixit02) [reply test/morris_method.jl]: Added.

**Comment by Vaibhavdixit02** [comment]
@ChrisRackauckas can you take a look at the version resolving issue? I can't recreate it locally, `test DiffEqSensitivity` gives me green flags for everything..

File: test/morris_method.jl (hunk 5 above)
Comment (ChrisRackauckas) [reply test/morris_method.jl]: test it simultaniously with linear here to make sure multi-output is fine?

File: test/morris_method.jl [hunk 6]
```diff
@@ -4,12 +4,54 @@ A = reshape([1,0,2,3],2,2)
...
+m = gsa(ishi, Morris(num_trajectory=10000), [[lb[i],ub[i]] for i in 1:4])
+@test m.means ≈ [ 2.23463  4.41884  2.5193  0.0] atol = 1e-2
+
+m = gsa(linear_batch,Morris(p_steps=[10,10],num_trajectory=10000),[[1,5],[1,5]],batch=true)
+@test m.means ≈ [7.0  0.1] atol = 1e-2
+
+m = gsa(ishi_linear,Morris(p_steps=[10,10],relative_scale=false,num_trajectory=100),[[lb[i],ub[i]] for i in 1:4])
+@test m.means[2,:] ≈ [7.0,0.1,0.0,0.0] rtol = 1e-12
+
+m = gsa(ishi_linear_batch,Morris(p_steps=[10,10],relative_scale=false,num_trajectory=100),[[lb[i],ub[i]] for i in 1:4])
```
Comment (ChrisRackauckas) [inline test/morris_method.jl]: ishi here too

**Comment by ChrisRackauckas** [comment]
Running locally it all passed.

**Pull request merged**